contracts/contracts/test-contracts/GasLimitTestContract.sol

```diff
@@ -0,0 +1,14 @@
+// SPDX-License-Identifier: AGPL-3.0
+pragma solidity 0.8.19;
+
+contract GasLimitTestContract {
+  uint256 internal _gasLimit;
+
+  function getGasLimit() external returns (uint256 gasLimit) {
+    return _gasLimit;
```

Comment:
you can assign `gasLimit = _gasLimit` vs. the return 

the current approach is mixing modes. Ideally just assign the return var.